src/schedule.rs

```diff
@@ -415,11 +456,22 @@ impl<Z> Iterator for OwnedScheduleIterator<Z> where Z: TimeZone {
     fn next(&mut self) -> Option<DateTime<Z>> {
         let previous = self.previous_datetime.take()?;
 
-        if let Some(next) = self.schedule.next_after(&previous) {
-            self.previous_datetime = Some(next.clone());
-            Some(next)
+        if let Some(later) = self.later_datetime.take() {
+            self.previous_datetime = Some(later.clone());
+            Some(later)
```

Comment:
Unfortunately, the `Copy` impl for `DateTime<Tz>` requires `<Tz as TimeZone>::Offset: Copy` and would require adding a number additional constraints in the cron code to be able to use.  I don't know if it's worth the change at the moment, considering it would be a semver breaking change.